jumpgate/identity/drivers/sl/tokens.py

```
@@ -70,11 +70,23 @@ class SLAuthDriver(identity.AuthDriver):
     def __init__(self):
         super(SLAuthDriver, self).__init__()
 
+    def _is_token_auth(self, username, credential, token_id):
+        return not username and not credential and token_id
+
     def authenticate(self, creds):
         username = lookup(creds, 'auth', 'passwordCredentials',
                           'username')
         credential = lookup(creds, 'auth', 'passwordCredentials',
                             'password')
+        token_id = lookup(creds, 'auth', 'token', 'id')
+        if self._is_token_auth(username, credential, token_id):
+            token = identity.token_id_driver().token_from_id(token_id)
+            try:
+                identity.token_driver().validate_token(token)
+                username = token['username']
+                credential = token['api_key']
+            except exceptions.InvalidTokenError as e:
+                return None
```

maybe best to not 'swallow' this exception here and return none... i would consider just letting the InvalidTokenError propigating up the stack. there is a mapper in `jumpgate.common.exceptions` which will transform this exception into an Unauthorized, but more importanatly it will dump the error to log before doing so to ease admin operations of this service.